.github/workflows/testing.yml

```diff
@@ -75,7 +74,48 @@ jobs:
       - name: Test
         env:
           MP_API_KEY: ${{ secrets.MP_API_KEY }}
-        run: pytest --cov=atomate2 --cov-report=xml
+        run: pytest --ignore=tests/forcefields --cov=atomate2 --cov-report=xml
+
+      - uses: codecov/codecov-action@v1
+        if: matrix.python-version == '3.10' && github.repository == 'materialsproject/atomate2'
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ./coverage.xml
+
+  test-forcefields:
```

Comment:
i'd recommend merging all tests back into a single job (which we can then split with `pytest-split`)

Reply:
It seems like anything `torch`-dependent and `tblite` can't be installed in the same environment without the `tblite` tests crashing in CI. If I use `pytest-split`, I'd still need to separately run the forcefield tests to have separate environments, no?

Outside of CI, having torch installed but not loaded seems OK with `tblite`. I can also just split off the `tblite` / `ase-ext` tests and then use `pytest-split` on the rest

Let me know what you think is best

Reply:
> It seems like anything torch-dependent and tblite can't be installed in the same environment without the tblite tests failing in CI.

oh, i forgot about that. in that case, could you verbatim copy your above comment into the `test.yml` to leave a paper trail on why the tests are split into 2 jobs? i do think it would be better to run the `tblite` in its own job and move the force fields into the main job. will put some pressure on me to actually implement the `pytest-split` to warrant the effort. 😅

Reply:
OK! Split off the ase and jupyter notebook tests into separate test instances. Also set up the `pytest-split` logic (3 splits per version seems OK), might need some help in getting that optimized

Reply:
`pymatgen` uses 10 splits. i think `atomate2` could use more than 3, maybe 5 is the sweet spot? how did you generate the durations file? ideally, all splits should take about the same amount of time. currently, 1 is ~35 min, 2 is 15-20 min and 3 is 5 - 10 min so i think there's something off. for comparison, `pymatgen` uses

```
pytest --store-durations --durations-path tests/files/.pytest-split-durations
```

to update the file

Reply:
I copied all the pytest split logic from pymatgen 😂 same command for the split durations test, the CI tests just take a very different amount of time in CI vs. my machine. Guessing there's MPS hardware acceleration turned on by default in the forcefields

I'll play around with this a bit - maybe setting the default device to be CPU would help for getting a better CI timing estimate

Reply:
the other option would be to temp change the CI run to

```sh
pytest --store-durations --durations-path=DURATIONS_PATH
cat DURATIONS_PATH
```

and then copy pasting the actual CI durations from the log into your local file and committing that